### DISCUSSION TITLE
SBIO Push Button

### DISCUSSION THREAD

**Discussion by D-Pyro** [MLB-LED-Scoreboard/mlb-led-scoreboard#470] [Ideas]
Add SBIO similar to NHL Scoreboard.

https://github.com/riffnshred/nhl-led-scoreboard/blob/master/src/sbio/SBIO.md

**Comment by ty-porter** [comment]
Looks like a big ask so I want to break down what I think. All this is barring hardware setup, just initial thoughts on integrating it with the codebase.

Dimmer: Seems the most straightforward to implement, given that it's just setting brightness on the matrix object
Pushbutton sleep/reboot/shutdown: Easy, doesn't need to live in this project, just use `sbin/reboot`/`sbin/shutdown` or "sleep" with dimmer set to 0% brightness

Now, for the others, things are much harder. We've identified screen rotation as being overly complex and having a bunch of tech debt in this project (see https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/issues/466) Without event-based architecture or some sort of way to manage yet another way to rotate through screens, I'm not sure it's feasible in the current state.

Things that would fall into that category are pushbutton rotate, screensaver (other than a simple black screen), or motion sensor (if you want it to trigger a screen on sensed motion and not just fade in/out from black).
